Core/CMP API Specification.md

```diff
@@ -325,13 +345,155 @@ A call to the `addEventListener` command must always respond with a return objec
   </tr>
 </table>
 
-______
-#### `removeEventListener` <a name="removeeventlistener"></a>
+The “signalStatus” event shall always be the first (if applicable) and last in a chain of events being fired by the CMP. 
 
-The `removeEventListener` command can be used to remove an existing event listener.
+A CMP **must** send all relevant events and it **must** send them in a specific order so that listeners (e.g. vendors) can understand when a task is completed. Whenever the CMP starts to change or is about to change any of the existing sections or is processing user input for an existing GPP string, it **must always** first set "signalStatus" to "not ready" and fire the corresponding event. It can then perform the tasks (e.g. change the sections along with firing the section change events). Only when all tasks are completed, the CMP can set the "signalStatus" to "ready" and fire the corresponding event.
 
+_Event Order Example 1:_ 
 
+The following example illustrates the  events that are fired, and the other order in which they are fired, assuming support for IAB TCF Canada for a new user.
+<div>
 <table>
+<tbody>
+<tr>
+<td><strong>Event Order</strong></td>
+<td><strong>Event Name</strong></td>
+<td><strong>Data</strong></td>
+<td><strong>Description</strong></td>
+</tr>
+<tr>
+<td>1</td>
+<td colspan="3"><em>initial data of PingReturn is {gppVersion:1.1, cmpStatus: loading, cmpDisplayStatus: hidden, signalStatus: not ready, ...}</em></td>
+</tr>
+<tr>
+<td>2</td>
+<td>listenerRegistered</td>
+<td>&nbsp;</td>
+<td>CMP has registered the event listener. Event is immediately fired after registering.</td>
+</tr>
+<tr>
+<td>3</td>
+<td>cmpStatus</td>
+<td>loaded</td>
+<td>CMP is now loaded. Event is fired with name=cmpStatus and data=loaded.</td>
+</tr>
+<tr>
+<td>4</td>
+<td>cmpDisplayStatus</td>
+<td>visible</td>
+<td>CMP now displays the consent layer. Event is fired with name=cmpDisplayStatus and data = visible</td>
+</tr>
+<tr>
+<td>5</td>
+<td colspan="3"><em>User makes their choices and clicks on accept or reject or save</em></td>
+</tr>
+<tr>
+<td>6</td>
+<td>cmpDisplayStatus</td>
+<td>hidden</td>
+<td>CMP closed the consent layer and processes user input. Event is fired with name=cmpDisplayStatus and data=hidden</td>
+</tr>
+<tr>
+<td>7</td>
+<td>sectionChange</td>
+<td>tcfcav1</td>
```

Comment:
appears incorrect. 

https://github.com/InteractiveAdvertisingBureau/Global-Privacy-Platform/blob/87935c4b574e974ff20ff5c1b9641374878031e2/Core/CMP%20API%20Specification.md?plain=1#L77 has tcfcav1 but the actual spec says https://github.com/InteractiveAdvertisingBureau/Global-Privacy-Platform/blob/87935c4b574e974ff20ff5c1b9641374878031e2/Sections/Section%20Information.md?plain=1#L46

Reply:
yes we had a bit of back and forth. i think we landed on "tcfcav1" as the correct spelling

Reply:
@janwinkler I see that is mentioned in #73 

We cannot merge this until this is fixed all over the project. This is a large mess and is breaking, so needs to be fixed before this is merged..

Reply:
my understanding was that @HeinzBaumann fixed this in one of his last pull requests

Reply:
It is not fixed

Reply:
Merged, thanks for flagging these.